### ISSUE TITLE
Returning empty object in @CrudAuth filter function results in invalid SQL

### ISSUE THREAD

**Issue by DaVarga** [nestjsx/crud#309]
Hey,

relating to #302 i found a bug. Returning a empty object in the filter function results in invalid SQL.

Something like
```ts
@CrudAuth({
  filter: req => {
    if(req.user.isRoot) {
      return {};
    }
    return {
      'trustedUsers.id': req.user.id,
    };
  },
})
```
will result in SQL like `SELECT [...] FROM "photo_entity" "PhotoEntity" WHERE  AND "PhotoEntity"."deletedAt" IS NULL ORDER BY "PhotoEntity"."id" DESC LIMIT 10` if `isRoot` is true and you try to add a filter via query params.

**Comment by zMotivat0r** [comment]
Thanks, but why would you do that?

**Comment by gugahoa** [comment]
@zMotivat0r Hi, I'm willing to take up this issue as it's blocking a feature on my day-job, could you provide some guidance on where do you think the bug is at?

**Comment by zMotivat0r** [comment]
fixed in `4.3.0-beta` versions
